driver-async/src/main/com/mongodb/async/client/OperationExecutorImpl.java

```diff
@@ -109,8 +111,9 @@ public void onResult(final ClientSession clientSession, final Throwable t) {
                         @Override
                         public void onResult(final T result, final Throwable t) {
                             try {
-                                if (t instanceof MongoSocketException && session != null && session.hasActiveTransaction()) {
-                                    ((MongoException) t).addLabel(MongoException.TRANSIENT_TRANSACTION_ERROR_LABEL);
+                                if (t instanceof MongoSocketException && session != null && session.hasActiveTransaction()
+                                        && !((MongoException) t).hasErrorLabel(UNKNOWN_TRANSACTION_COMMIT_RESULT_LABEL)) {
```

Comment:
Spec requires that TRANSIENT_TRANSACTION_ERROR_LABEL is added except whenUNKNOWN_TRANSACTION_COMMIT_RESULT_LABEL is present. It's necessary to do it this way because UNKNOWN_TRANSACTION_COMMIT_RESULT_LABEL is added by CommitTransactionOperation, prior to this check